enhancements/opm-content-management.md

```diff
@@ -0,0 +1,147 @@
+---
+title: opm-content-management 
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# OPM Content Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This enhancement is a proposal for extending the existing capabilites of the `opm` command line utility, to provide more fine grained control over the management of catalog indexes.   
+
+`opm` will significantly improve the UX around working with indexes by allowing removal of individual bundles from a package and removal of channels from a package. Indexes will now also be able to be queried directly for inspection, and update graph visualization for individual packages will be added. Bundle validation without a deamon by default is also proposed in this enhancement, keeping docker users in mind. 
+
+## Motivation
+
+As more and more operators are migrated to the new bundle format, and more index images are created for Operator Catalogs, creators of catalogs of operators, such as the Red Hat pipeline responsible for maintaining the `redhat-operators` catalog, should have more fine grained control over the management of the catalogs using the `opm` tooling.  
+
+### Goals
+
+- Allow `opm` users to perform bundle validations in environments without a Docker daemon. Bundle validation also includes validation of bundle image labels against `metadata/annotations.yaml`
+- Allow `opm` users to safely edit indexes, which includes removing bundle version from a package in an index, and removing a channel from a bundle. 
+- Allow `opm` users to inspect catalog content and return the update graph of Operators in machine-readable format.
+
+### Non-Goals
+
+
+## Proposal
+ 
+### Implementation Details
+
+### Validate the bundle image labels against bundle annotations
+
+Currently, `opm alpha bundle validate` validates only the content of `metadata/annotations.yaml` file, but not the image labels of the DockerFile. The image labels are used when the bundle is added to an index image. As a result, if there are mismatches between the content of the bundle image (i.e `metadata/annotations.yaml`) and the bundle image labels, the bundle image is invalid and could potentially cause issues when it is added to an index image.
+
+The `opm alpha bundle validate` command will be enhanced with an extra step that will check to see if the contents of `metadata/annotations.yaml` matches with the labels of the bundle image. If the `metadata/annotations.yaml` contain fields that are not there in the labels of the image or vice versa, the validation will fail.  
```

Comment:
Are we saying we want to fail if the set of labels in the `annotations.yaml` don't exactly match the labels on the generated image? I believe what we want is the image labels to be a superset of the the `annotation.yaml` labels, so they can be customized and used in an ad-hoc way (they are just images). 

I think this is what you're saying here, but maybe just clarify a bit because it is a bit confusing

Reply:
@exdx aren't the labels in the images supposed to be exactly the annotations in `annotations.yaml`?

Reply:
I think we want to fail if there are labels in `annotations.yaml` that are not on the image. I think other tooling/registries should remain able to apply additional labels on the image without `opm bundle validate` complaining.